docs/howto/query.md

```diff
@@ -0,0 +1,225 @@
+
+### Get all the rows
+To get all the rows returned from the database use `Query.GetAll`, passing
+pointers to slices of all the output variables in the query.
+
+For example:
+```go
+location := Location{City: "Edinburgh"}
+var employees []Employee
+
+err := db.Query(ctx, stmt, location).GetAll(&employees)
+if err != nil {
+    return err
+}
+
+// employees now contains all the employees returned from the database.
+```
+
+> See more:
+[`Query.GetAll`](https://pkg.go.dev/github.com/canonical/sqlair#Query.GetAll)
+
+### Iterate over the rows
+To iterate over the rows returned from the query, get an `Iterator` with use
+`Query.Iter`.
+
+`Iterator.Next` prepares the next row for `Iterator.Get`. It will return false
+if there are no more rows or there is an error. `Iterator.Get` works the same as
+`Query.Get` above, except it gets the current row.
+
+Make sure to run `Iterator.Close` once you are finished iterating. The
+`Iterator.Close` operation should generally be deferred when the `Iterator` is
+created. Any errors encountered during iteration will be returned with
+`Iterator.Close`.
+
+For example:
+```go
+location := Location{City: "Edinburgh"}
+
+iter := db.Query(ctx, stmt, location).Iter()
+
+// Defer closing of the iterator and set its error to the error returned from 
+// the function (if the function error is not nil).
+defer func(){
+    closeErr := iter.Close()
+    if err == nil {
+        err = closeErr
+    }
+}()
+
+for iter.Next() {
+    var employee Employee
+    err := iter.Get(&employee)
+    if err != nil {
+        return err
+    }  
+}
+```
+
+> See more:
+[`Query.Iter`](https://pkg.go.dev/github.com/canonical/sqlair#Query.Iter),
+[`sqlair.Iterator`](https://pkg.go.dev/github.com/canonical/sqlair#Iterator),
+[`Iterator.Next`](https://pkg.go.dev/github.com/canonical/sqlair#Iterator.Next),
+[`Iterator.Get`](https://pkg.go.dev/github.com/canonical/sqlair#Iterator.Get),
+[`Iterator.Close`](https://pkg.go.dev/github.com/canonical/sqlair#Iterator.Close)
+### Just run 
+To run a query that does not return any rows, use `Query.Run`. This is useful
+when doing operations that are not expected to return anything.
+
+For example:
+```go
+stmt, err := sqlair.Prepare("INSERT INTO employee (*) VALUES ($Employee.*)", Employee{})
+if err != nil {
+    return err
+}
+
+employee := Employee{
+    ID:         1 
+    Name:       "Joe"
+    LocationID: 17
+}
+err := tx.Query(ctx, stmt, employee).Run()
+if err != nil {
+    return err
+}
+// employee has been inserted into the database.
+```
+
+> See more:
+[`Query.Run`](https://pkg.go.dev/github.com/canonical/sqlair#Query.Run)
+
+
+## 5. (Optional) Get the query outcome
+
+To get the outcome, provide a pointer to a `sqlair.Outcome` as the first
+argument to any of the `Get` methods. This will fill the `sqlair.Outcome` with
+information about the outcome of the query.
+
+The query outcome contains metadata about the execution of a query. Currently,
```

Comment:
This feels like material for a note (BTW sort of content).

Reply:
Nice, good point. Notes always make this sort of thing look more professional too